### PR TITLE
support for orthography profile handling

### LINKED ISSUE #207: Mark in FormTable which profile a segmentation was based on
Since we support tokenization based on multiple profiles per dataset, it would be good to be able to trace back which profile was used for which form - e.g. to compute frequencies in the profile linter.

This could be done by adding a `Profile` column to `FormTable`. But it isn't fully clear what values to use for this column. One candidate would be the keys of the tokenizers `dict`, i.e. the stems of filenames in `etc/orthography`. Full paths might be more transparent, but would also blow up the forms table.
So I'd propose:
- `False`: no profile used. Segments have been provided in another way.
- `True`: segmented via `etc/orthography.tsv`
- `<stem>`: segmented via `etc/<stem>.tsv`

### PR DESCRIPTION
**Pull request by xrotwang**
first stab at #169

closes #207

### CODE REVIEW

File: src/pylexibank/commands/format_profile.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+
+def unicode2codepointstr(text):
+    """
+    Returns a codepoint representation to an Unicode string.
+    """
+    return " ".join(["U+{0:0{1}X}".format(ord(char), 4) for char in text])
+
+
+def check_consistency(profile, clts, args):
```
Comment (xrotwang) [inline src/pylexibank/commands/format_profile.py:90]: @tresoldi `check_consistency` and `clean_profile` will be moved to methods of `pylexibank.profile.Profile` as well.

File: tests/test_profile.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+def test_sort():
+    prf = Profile({'Grapheme': 'ab', 'IPA': 'z'}, {'Grapheme': 'x', 'IPA': 'y'})
+    assert list(prf.graphemes.keys())[0] == 'ab'
+    prf.sort()
+    assert list(prf.graphemes.keys())[0] == '^'
+    assert list(prf.graphemes.keys())[-1] == 'ab'
+
+
+def test_trim():
```
Comment (xrotwang) [inline tests/test_profile.py:17]: @tresoldi If you have ideas for more tests, let me know!

File: src/pylexibank/commands/format_profile.py (hunk 1 above)
Comment (tresoldi) [reply src/pylexibank/commands/format_profile.py:90]: :+1:

File: tests/test_profile.py (hunk 2 above)
Comment (tresoldi) [reply tests/test_profile.py:17]: I could add a couple, if you prefer. The most important thing is to check if it is handling all cases with slash notation correctly (I think there might be a bug in my code for cases where the left side is empty, for example).

Alternatively, I can just work on one more profile and will probably hit all cases that would deserve testing. :wink:

**Comment by xrotwang** [comment]
@tresoldi I'm happy with the functionality right now, so will merge. And yes, integration testing by using `cldfbench lexibank.format_profile` on real-world datasets is very welcome!

**Comment by tresoldi** [comment]
@xrotwang I will start checking a fiddling around with it. Should we later, in a week or two, even consider raising warning in buildbot? For example if the result of `format_profile` is different from the one in the repository?

**Comment by xrotwang** [comment]
@tresoldi If we can agree on the exact command, then yes. Dunno how much agreement there is on the particular sorting and possibly on the effects of trimming, though. I think buildbot should mainly make sure the CLDF data is as good as possible, and only inspect auxiliary files in `etc/` as far as this is relevant for the quality of the CLDF data.

**Pull request merged**